### DISCUSSION TITLE
MSEgui version 5.10.4.

### DISCUSSION THREAD

**Discussion by fredvs** [mse-org/mseide-msegui#85] [General]
![msegui_24](https://github.com/mse-org/mseide-msegui/assets/3421249/c8899a26-0072-467c-8364-4d01cda2ea1a)  **MSEide+MSEgui Pascal Cross Platform GUI Development System**.

2024-04-13 MSEgui version **5.10.4**.

 Copyright (c) 1999-2024 by Martin Schreiber and friends.

- Compiles with FPC 2.6.4, 3.0.0, 3.0.2, 3.2.0, 3.2.2 and 3.3.1.
- Compiles with FPC-LLVM 3.3.1.
- For Linux-x86_64, Linux-i386,
  Linux-Rpi-arm32, Linux-Rpi-aarch64,
  FreeBSD-x86_64, FreeBSD-i386,
  NetBSD-x86_64, NetBSD-i386,
  OpenBSD-x86_64, OpenBSD-i386, 
  Darwin-MacOs-x86_64,  Darwin-MacOs-aarch64,
  Windows-i386, Windows-x86_64.
- Links to xlib and gdi32, no external widget library needed.
- Internal character encoding is utf-16.
- Uses anti aliased fonts on Linux (Xft).
- All screen drawing is double buffered.
- Has docking forms and MDI.
- Has embedded forms (similar to TFrame).
- Has a class bridge that can be enabled.
- Has sophisticated database access components and data edit widgets.
- Internationalization functionality with resource modules.
- Report generator.
- BGRABitmap graphic library compatible.

IDE:
- Integrated debugging.
- Source code highlighting.
- Source code navigation with support for include files.
- Code completion for classes.
- Procedures list.
- Integrated visual form designer with source code update for components and 
  events.
- Flexible and handy build system with switchable macros.
- Visual form inheritance.
- Integrated report designer.
- Classes bridge provided.

Video of installation + load/run demo: 
https://user-images.githubusercontent.com/3421249/198116838-e3eaa3ca-0150-4df5-8c9f-b23bd945f5ba.mp4

**_Warning_**: For Windows, with optimization **>= 3**, better use  -**OoNODFA** to disable **DFA**.  See here: https://github.com/mse-org/mseide-msegui/discussions/60

**_Warning_**: For Darwin-MacOs, you need to install **xquartz** and **fontconfig**:
                   `> brew install xquartz fontconfig`

For the release of binary **MSEide**, please use the assets of release 5.10.0. (no change in MSEide for MSEgui 5.10.4.)
See assets at end of page:
https://github.com/mse-org/mseide-msegui/releases/tag/5.10.0

<hr /><em>This discussion was created from the release <a href='https://github.com/mse-org/mseide-msegui/releases/tag/5.10.4'>MSEgui version 5.10.4.</a>.</em>

**Comment by fredvs** [comment]
This release 5.10.4 fixes mselibc.pas for Linux i386 and when fpc used <= version 3.0.2.
See here: https://forum.lazarus.freepascal.org/index.php/topic,55217.msg514026.html?PHPSESSID=41sd602fpnus8p7bm621igr9j4#msg514026

and here:

https://forum.lazarus.freepascal.org/index.php/topic,55217.msg514181.html?PHPSESSID=41sd602fpnus8p7bm621igr9j4#msg514181

**Comment by rchastain** [comment]
Good job Fred!

Below my script for downloading, building and installing version 5.10.4 under Linux.

[installmseide5104.zip](https://github.com/mse-org/mseide-msegui/files/14967691/installmseide5104.zip)

**Reply by fredvs** [reply]
Ho, very nice,  that installer should be more visible and also added in the readme.md.

**Reply by fredvs** [reply]
Your script also compiles MSEide 5.10. using MSEgui 5.10.4.
Perfect.
Would be also very nice to have a Windows installer.
